pkg/icingadb/ha.go

```diff
@@ -274,15 +287,25 @@ func (h *HA) realize(ctx context.Context, s *icingaredisv1.IcingaStatus, t *type
 			switch errQuery {
 			case nil:
 				otherResponsible = true
-				if shouldLog {
+				if shouldLogRoutineEvents {
 					h.logger.Infow("Another instance is active",
 						zap.String("instance_id", instance.Id.String()),
 						zap.String("environment", envId.String()),
-						"heartbeat", instance.Heartbeat,
+						zap.Time("heartbeat", instance.Heartbeat.Time()),
 						zap.Duration("heartbeat_age", time.Since(instance.Heartbeat.Time())))
 				}
 			case sql.ErrNoRows:
-				takeover = true
+				fields := []any{
+					zap.String("instance_id", instance.Id.String()),
+					zap.String("environment", envId.String()),
+					zap.Duration("heartbeat_timeout", timeout)}
+
+				if !h.responsible {
+					h.logger.Infow("Preparing to take over HA as no instance is active", fields...)
```

Comment:
Is this even necessary when we're going to log `Taking over ...` in the main function just in the next line? I think it would be sufficient to just include the environment ID as part of the `Taking over ...` log line and then remove this completely.

Reply:
This will look like the following:

```
icingadb-master2  | 2024-03-14T11:24:05.914Z    INFO    high-availability       Preparing to take over HA as no instance is active      {"envir
onment": "cef9c044b80d8cf516b031c0162ef41394b36498", "heartbeat_timeout": "1m0s"}
icingadb-master2  | 2024-03-14T11:24:05.919Z    INFO    icingadb        Taking over     {"reason": "no other instance is active"}
```

Yes, this could be seen as a duplicate, but in the rare case that between making the decision to take over and actually taking over something breaks, we would know on which code path Icinga DB was.

Maybe it could be reduced to debug logging?

Reply:
After giving this a second thought, I would degrade this to `DEBUG`. Same as the two sibling messages "Another instance is active" and "Continuing being the active instance" as they are, by definition, only useful for debugging.

I would like to write the second part to <https://github.com/Icinga/icingadb/pull/692#discussion_r1528720666>, but GitHub does not show me a text box to comment below this comment.